RTSP panic: stream cannot be different than the one returned in previous OnSetup call (gortsplib v5.2.0)

### Which version are you using?

1.15.4

### Which operating system are you using?

Linux amd64 Docker

### Describe the issue

**Behavior / frequency**
From time to time Mediamtx crashes with a panic coming from gortsplib:
`panic: stream cannot be different than the one returned in previous OnSetup call`
The crash is not deterministic: sometimes it happens roughly once per hour, and sometimes the server can run up to ~6 hours without issues. After the panic, the Mediamtx process exits.

**Environment**
- Mediamtx version: (please fill in exact tag or commit, e.g. v1.9.x / Docker image tag you’re using)
- gortsplib version: v5.2.0 (from stack trace)
- OS: (for example: Ubuntu 22.04 / Debian / Docker base image)
- Architecture: amd64 (x86_64) / arm64 (please fill in)
- Deployment: Docker / bare metal / other (please specify)

**What clients are doing**

- RTSP is enabled with TCP-only transport (rtspTransports: [tcp]).
- There are RTSP clients that publish / play streams over RTSP.
- I do not explicitly manipulate RTSP sessions or OnSetup callbacks myself; all is handled by Mediamtx internally.

**Expected behavior**

- Mediamtx should not panic, even if a client misbehaves or sends unexpected RTSP requests. 
- In the worst case, the offending session should be closed, but the server should keep running.

### Describe how to replicate the issue


The panic occurs sporadically:

- sometimes after about 1 hour of operation,
- sometimes the server runs up to ~6 hours without any issues.

When the panic happens, Mediamtx exits, and I need to restart the process.

### MediaMTX configuration

```
logLevel: info

rtsp: yes
rtspTransports: [tcp]
rtspAddress: :8554
rtmpAddress: :1935
hlsAddress: :8888

api: yes
apiAddress: :9997
metricsAddress: :9998
srtAddress: :8890

playback: yes
playbackAddress: :9996

webrtc: yes
webrtcAddress: :8889
webrtcEncryption: no
webrtcAllowOrigin: "*"
webrtcLocalUDPAddress: :8189
webrtcAdditionalHosts: []
webrtcICEServers2: []

authInternalUsers:
  - user: any
    pass: ""
    ips: ['172.0.0.0/8', '192.168.0.1/8']
    permissions:
      - action: api
      - action: playback

pathDefaults:
  record: no
  recordPath: /recordings/%path/%Y-%m-%d_%H-%M-%S-%f
  recordFormat: fmp4
  recordPartDuration: 1s
  recordMaxPartSize: 10M
  recordSegmentDuration: 5s
  recordDeleteAfter: 20m

```

### MediaMTX logs

```
panic: stream cannot be different than the one returned in previous OnSetup call

goroutine 55101 [running]:
github.com/bluenviron/gortsplib/v5.(*ServerSession).handleRequestInner(0xc001eaca80, 0xc004a81e10, 0xc001a6d580)
/go/pkg/mod/github.com/bluenviron/gortsplib/v5@v5.2.0/server_session.go:1164 +0x3238
github.com/bluenviron/gortsplib/v5.(*ServerSession).runInner(0xc001eaca80)
/go/pkg/mod/github.com/bluenviron/gortsplib/v5@v5.2.0/server_session.go:776 +0x350
github.com/bluenviron/gortsplib/v5.(*ServerSession).run(0xc001eaca80)
/go/pkg/mod/github.com/bluenviron/gortsplib/v5@v5.2.0/server_session.go:723 +0x114
created by github.com/bluenviron/gortsplib/v5.(*ServerSession).initialize in goroutine 37
/go/pkg/mod/github.com/bluenviron/gortsplib/v5@v5.2.0/server_session.go:378 +0x391
```

### Network dump

_No response_

--- a/internal/servers/rtsp/session.go
+++ b/internal/servers/rtsp/session.go
@@ -194,6 +194,13 @@ func (s *session) onAnnounce(c *conn, ctx *gortsplib.ServerHandlerOnAnnounceCtx)
 	}, nil
 }
 
+func (s *session) rtspStream() *gortsplib.ServerStream {
+	if !s.isTLS {
+		return s.stream.RTSPStream(s.rserver)
+	}
+	return s.stream.RTSPSStream(s.rserver)
+}
+
 // onSetup is called by rtspServer.
 func (s *session) onSetup(c *conn, ctx *gortsplib.ServerHandlerOnSetupCtx,
 ) (*base.Response, *gortsplib.ServerStream, error) {
@@ -226,7 +233,7 @@ func (s *session) onSetup(c *conn, ctx *gortsplib.ServerHandlerOnSetupCtx,
 	}
 
 	switch s.rsession.State() {
-	case gortsplib.ServerSessionStateInitial, gortsplib.ServerSessionStatePrePlay: // play
+	case gortsplib.ServerSessionStateInitial: // play
 		path, stream, err := s.pathManager.AddReader(defs.PathAddReaderReq{
 			Author: s,
 			AccessRequest: defs.PathAccessRequest{
@@ -261,16 +268,14 @@ func (s *session) onSetup(c *conn, ctx *gortsplib.ServerHandlerOnSetupCtx,
 		s.path = path
 		s.stream = stream
 
-		var rstream *gortsplib.ServerStream
-		if !s.isTLS {
-			rstream = stream.RTSPStream(s.rserver)
-		} else {
-			rstream = stream.RTSPSStream(s.rserver)
-		}
-
 		return &base.Response{
 			StatusCode: base.StatusOK,
-		}, rstream, nil
+		}, s.rtspStream(), nil
+
+	case gortsplib.ServerSessionStatePrePlay: // play, subsequent calls
+		return &base.Response{
+			StatusCode: base.StatusOK,
+		}, s.rtspStream(), nil
 
 	default: // record
 		return &base.Response{

